tests/sqlparser_common.rs

```diff
@@ -11432,3 +11432,15 @@ fn test_any_some_all_comparison() {
     verified_stmt("SELECT c1 FROM tbl WHERE c1 <> SOME(SELECT c2 FROM tbl)");
     verified_stmt("SELECT 1 = ANY(WITH x AS (SELECT 1) SELECT * FROM x)");
 }
+
+#[test]
+fn test_alias_equal_expr() {
+    let dialects = all_dialects_where(|d| d.supports_eq_alias_assigment());
```

Comment:
@yoavcloud just one comment on the test [from here](https://github.com/apache/datafusion-sqlparser-rs/pull/1467#discussion_r1798900193) if we can include a similar case for the other dialects verifying that the `some_alias = some_column` gets parsed as an actual expression vs an alias?

Beyond that I think this should be good to go, thanks!

Reply:
Good idea, done!